ompi/mpi/c/wtime.c

```diff
@@ -40,17 +44,30 @@ double MPI_Wtime(void)
 {
     double wtime;
 
+    /*
+     * See https://github.com/open-mpi/ompi/issues/3003
+     * For now we are forcing the use of gettimeofday() until we find a
+     * more portable solution.
+     */
+#if 0
 #if OPAL_TIMER_CYCLE_NATIVE
     wtime = ((double) opal_timer_base_get_cycles()) / opal_timer_base_get_freq();
 #elif OPAL_TIMER_USEC_NATIVE
     wtime = ((double) opal_timer_base_get_usec()) / 1000000.0;
+#endif
+#else
+#if defined(__linux__) && OPAL_HAVE_CLOCK_GETTIME
+    struct timespec tp = {.tv_sec = 0, .tv_nsec = 0};
+    (void) clock_gettime(CLOCK_MONOTONIC, &tp);
+    return (tp.tv_sec + tp.tv_nsec/1.0e+9);
```

Comment:
This `return` is not quite right -- all paths through the original code assign `wtime` and fall through to call `OPAL_CR_NOOP_PROGRESS`.

I know the checkpoint/restart code has fallen into disrepair, but the disparity of treatment here is a little odd.